src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -145,6 +147,9 @@ public String getLabel() {
          * @param r The instance to compute Java options for
          */
         private void computeJavaOptions(RealJenkinsRule r) {
+            if (cert != null || noCertificateCheck) {
+                return;
+            }
             if (trustStorePath != null && trustStorePassword != null) {
```

Comment:
Ah, did not realize this was already here. So Javadoc should clarify that these options are mutually exclusive.

Reply:
agree. updated now. ty